python/lsst/analysis/ap/nb_utils.py

```diff
@@ -70,3 +78,203 @@ def make_simbad_link(ra, dec, radius_arcsec=3.0):
+    matched: `pandas.DataFrame`
+        Data frame of matched sources; actually only the sources from
+        the first dataset but with a new column pointing to the
+        diaSourceId of the match in the decond dataset.
+    """
+
+    butler1 = dafButler.Butler(repo1, collections=coll1)
+    butler2 = dafButler.Butler(repo2, collections=coll2)
+
+    if badFlagList is not None:
+        query1.set_excluded_diaSource_flags(badFlagList)
+        query2.set_excluded_diaSource_flags(badFlagList)
+    goodSrc1 = query1.load_sources(exclude_flagged=True)
+    goodSrc2 = query2.load_sources(exclude_flagged=True)
+
+    if 'reliability' not in goodSrc1.columns:
+        goodSrc1['reliability'] = None
+    if 'reliability' not in goodSrc2.columns:
+        goodSrc2['reliability'] = None
+
+    """
+    We assume that the runs to be compared here will always be
+    over the same dataset with the same visit and detector numbers.
+    Then we can enforce the matching on that level, too.
+    We could relax that assumption later, though it would require
+    removing the visit, detector loop.
+    """
+
+    visit_det = set(zip(goodSrc1['visit'], goodSrc1['detector']))
+    src1 = []
+    src2 = []
+    matches = []
+    indices = {}
+    seps = {}
+
+    for visit, detector in visit_det:
+        mask1 = (goodSrc1['visit'] == visit) &\
+            (goodSrc1['detector'] == detector)
+        mask2 = (goodSrc2['visit'] == visit) &\
+            (goodSrc2['detector'] == detector)
+        gs1 = goodSrc1[mask1].copy()
+        gs2 = goodSrc2[mask2].copy()
+
+        coords1 = SkyCoord(ra=gs1['ra'].values*u.degree,
+                           dec=gs1['decl'].values*u.degree)
+        coords2 = SkyCoord(ra=gs2['ra'].values*u.degree,
+                           dec=gs2['decl'].values*u.degree)
+
+        index, sep, _ = match_coordinates_sky(coords1, coords2)
+        indices[(visit, detector)] = index
+        seps[(visit, detector)] = sep
+        gs1['xmatch_dist_arcsec'] = sep.to_value(u.arcsecond)
+        gs1['src2_diaSourceId'] = gs2['diaSourceId'].values.astype(np.int64)[index]
+
+        # Set the match ID to 0 if the distance is above threshold
+        gs1.loc[(gs1['xmatch_dist_arcsec'] > matchRadius),
+                ['src2_diaSourceId']] = 0
+
+        # get the DiaSources in dataset 2 not matched to something in dataset 1
+        uniqueid2 = set(gs2['diaSourceId']) - set(gs1['src2_diaSourceId'])
+        unique2 = gs2[gs2['diaSourceId'].isin(uniqueid2)]
+
+        unique1 = gs1[(gs1['src2_diaSourceId'] == 0)]
+
+        withmatch = gs1[(gs1['src2_diaSourceId'] > 0)]
+
+        src1.append(unique1)
+        src2.append(unique2)
+        matches.append(withmatch)
+
+    # Out of the loop, concatenate everything together.
+    unique1 = pd.concat(src1)
+    unique2 = pd.concat(src2)
+    matched = pd.concat(matches)
+
+    # Make cutouts if they don't already exist; optionally display to screen
```

Comment:
Creating cutouts can take a very long time, so please add an option to skip the code below and just return the dataframes of sources.